packages/apps/human-app/frontend/src/api/services/worker/my-jobs-data.ts

```diff
@@ -9,12 +9,44 @@ import { createPaginationSchema } from '@/shared/helpers/create-pagination-schem
 import type { MyJobsFilterStoreProps } from '@/hooks/use-my-jobs-filter-store';
 import { useMyJobsFilterStore } from '@/hooks/use-my-jobs-filter-store';
 
+export enum MyJobStatus {
+  ACTIVE = 'ACTIVE',
+  CANCELED = 'CANCELED',
+  COMPLETED = 'COMPLETED',
+  VALIDATION = 'VALIDATION',
+  EXPIRED = 'EXPIRED',
+  REJECTED = 'REJECTED',
+}
+
+// Repeated because of linting rule
```

Comment:
FYI: linting rule prevented me from creating `MyJobStatusWithUnknown` with all the values based directly on the `MyJobStatus`. The `UNKNOWN` is a type for handling unknown statuses on frontend, hence why it's not in the filters list. 
Using utility types was not an option as there were places where `MyJobStatus` was used as a value.

Reply:
@mpblocky 
> The UNKNOWN is a type for handling unknown statuses on frontend, hence why it's not in the filters list.

That's legit, we don't need this value in filters list. At the same time, we don't need to have a second enum.
You can have `MyJobStatus` enum as it as and the define next:

```
const UNKNOWN_JOB_STATUS = 'UNKNOWN';

export type MyJobStatusWithUnknown =
  | `${MyJobStatus}`
  | typeof UNKNOWN_JOB_STATUS;
```
in case you want exact types.

At the same time, the only place where you need it - is `getChipStatusColor`, so instead of having this `MyJobStatusWithUnknown`, you can simply change `getChipStatusColor` to accept `status: string` and return default value when not active|completed|validation. Right now this `getChipStatusColor` look like a common utility that can be used anywhere else because it's in shared helpers, but why we use `MyJobStatusWithUnknown` type then? It makes more sense to remove enum usage here and use hardcoded value if it's shared utility.

If we don't plan to have it like shared - let's move it some other place close to "my jobs" and keep `MyJobStatusWithUnknown` with unknown for it and `throw` in `default` block of switch-case statement

Reply:
This way we have two options, as we are using this enum in here:
`const buttonDisabled = d.status !== MyJobStatusWithUnknown.ACTIVE;`

1. delete `MyJobStatusWithUnknown` altogether and just use magic string `'UNKNOWN'` along with `MyJobsStatus` which will result in:
` const buttonDisabled = d.status !== MyJobStatus.ACTIVE && d.status !== UNKNOWN_JOB_STATUS;`

2. Keep the current solution.

Which would You prefer?

About the plan I'm not sure if we are going the use the chip anywhere else so it's entirely Your decision too :sweat_smile:

Reply:
@mpblocky option 1, but I don't get why we need to check for `d.status !== UNKNOWN_JOB_STATUS;`?

Button should be enabled only if `d.status === MyJobStatus.ACTIVE`, otherwise disabled (which is just `d.status !== MyJobStatus.ACTIVE`), so you don't have to check for unknown

Reply:
True, now it works - must have something broken with my eslint since it complained about no overlap :thinking: Will do the 1st version